tokio/src/runtime/task/mod.rs

```diff
@@ -201,10 +204,36 @@ use std::{fmt, mem};
 /// [unstable]: crate#unstable-features
 #[cfg_attr(docsrs, doc(cfg(all(feature = "rt", tokio_unstable))))]
 #[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
-// TODO(eliza): there's almost certainly no reason not to make this `Copy` as well...
-#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq)]
 pub struct Id(u64);
 
+/// Returns the id of the currently running task.
+///
+/// # Panics
+///
+/// This function panics if called from outside a task. Please note that calls
+/// to `block_on` do not have task ids, so the method will panic if called from
+/// within a call to `block_on`. For a version of this function that doesn't
+/// panic, see [`task::try_id()`](crate::runtime::task::try_id()).
+///
+#[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
+#[track_caller]
+pub fn id() -> Id {
+    use crate::runtime::context;
+    context::current_task_id().expect("Can't get a task id when not inside a task")
+}
+
+/// Returns the id of the currently running task. This function is similar to
+/// [`task::id()`](crate::runtime::task::id()), except it doesn't panic and
+/// returns `None` if the id is unavailable.
+///
+#[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
```

Comment:
let's add the unstable API note at the end of the docs here:
```suggestion
///
/// **Note**: This is an [unstable API][unstable]. The public API of this type
/// may break in 1.x releases. See [the documentation on unstable
/// features][unstable] for details.
///
/// [task ID]: crate::task::Id
#[cfg_attr(not(tokio_unstable), allow(unreachable_pub))]
```